pkg/uroot/uroot.go

```diff
@@ -140,21 +140,28 @@ func CreateInitramfs(opts Opts) error {
 
 	// Add files from command line.
 	for _, file := range opts.ExtraFiles {
+		var src, dst string
 		parts := strings.SplitN(file, ":", 2)
-		path, err := filepath.Abs(filepath.Join(parts...))
-		if len(parts) != 2 {
-			parts = append(parts, parts[0][1:])
+		if len(parts) == 2 {
+			// treat the entry with the new src:dst syntax
+			src = parts[0]
```

Comment:
no, this is not right. Consider:
/path/to/my/root:etc/hosts
src is /path/to/my/root/etc/hosts
dest is /etc/hosts
The code probably should check the argument more carefully, I just figured people would get an error and
rerun the command with proper arguments.

Reply:
not sure I understand - probably because my assumptions on the intended syntax are different. Trying to clarify. To reuse your example, we call `u-root -files /path/to/my/root:etc/hosts`. To me this means "I want the file /path/to/my/root to be created as /etc/hosts in the rootfs". But from your example I assume that you're using the colon as a regular file name component (i.e. "root:etc" is a directory under /path/to/my, is that what you mean?).

So if I get it correctly you want to handle file paths containing colon symbols. However as far as I know, ext* filesystems allow for any byte except null to be part of a file path, so I guess we can't be entirely generic and some limitation has to be accept (and documented).
Another option for the colon could be to prepend a character to each file path to tell which type of syntax to use, but I'd rather not go with it as it is counter-intuitive and can quickly become cumbersome.

At the moment there's already another big limitation, i.e. spaces. The current syntax allows for spaces as separators (e.g. `-files "src1:dst1 src2:dst2"` - it's done with `strings.Fields`) but the space is a valid path character.

Which option would you go with?

Reply:
: is a separator, not a part of the pathname.


it was not a regular pathname component.

That said I like your approach better, it's really more useful